sources/xfscli/src/bin/xfs_info.rs

```diff
@@ -0,0 +1,61 @@
+use std::process::exit;
+use std::process::Command;
+use std::str;
+
+use argh::FromArgs;
+
+/// display XFS filesystem geometry information
+#[derive(FromArgs)]
+struct Args {
+    /// print the version
+    #[argh(switch, short = 'V')]
+    version: bool,
+    /// mount-point | block-device | file-image
+    #[argh(positional)]
+    target: Option<String>,
+}
+
+static XFS_SPACEMAN: &str = "/usr/sbin/xfs_spaceman";
+static FINDMNT: &str = "findmnt";
+static XFS_DB: &str = "/usr/sbin/xfs_db";
+
+fn main() {
+    let args: Args = argh::from_env();
+
+    // Return the version for arg version as true
+    if args.version {
+        let xfs_spaceman_result = Command::new(XFS_SPACEMAN)
+            .args(["-p", "xfs_info", "-V"])
+            .output()
+            .expect("Couldn't run xfs_spaceman for version info");
+        print!("{}", String::from_utf8(xfs_spaceman_result.stdout).unwrap());
+        exit(0);
+    }
+
+    if let Some(target) = &args.target {
+        // Check if target is mounted
+        let mountpt = Command::new(FINDMNT)
+            .args(["-t", "xfs", "-f", "-n", "-o", "TARGET", target.as_str()])
+            .output()
+            .expect("Couldn't find the mount point");
+
+        // Get info using XFS_SPACEMAN if mounted else from XFS_DB
+        if mountpt.status.success() {
+            let mpt = String::from_utf8(mountpt.stdout).unwrap();
+            let xfs_spaceman_result = Command::new(XFS_SPACEMAN)
+                .args(["-p", "xfs_info", "-c", "info", mpt.as_str().trim()])
+                .output()
+                .expect("Couldn't run xfs info for mounted point");
+
+            print!("{}", String::from_utf8(xfs_spaceman_result.stdout).unwrap());
+        } else {
+            let xfs_spaceman_result = Command::new(XFS_DB)
+                .args(["-p", "xfs_info", "-c", "info", target.as_str()])
+                .output()
+                .unwrap();
+            print!("{}", String::from_utf8(xfs_spaceman_result.stdout).unwrap());
+        }
+    } else {
+        println!("Couldn't parse target  mount-point | block-device | file-image");
```

Comment:
This should exit 2 to stay consistent with https://git.kernel.org/pub/scm/fs/xfs/xfsprogs-dev.git/tree/spaceman/xfs_info.sh